mongo/cursor.go

```diff
@@ -186,10 +186,18 @@ func (c *Cursor) Close(ctx context.Context) error {
 func (c *Cursor) All(ctx context.Context, results interface{}) error {
 	resultsVal := reflect.ValueOf(results)
 	if resultsVal.Kind() != reflect.Ptr {
-		return errors.New("results argument must be a pointer to a slice")
+		return errors.New("results argument must be a pointer to a slice, but was a" + resultsVal.Kind().String())
 	}
 
 	sliceVal := resultsVal.Elem()
+	if sliceVal.Kind() == reflect.Interface {
+		sliceVal = sliceVal.Elem()
+	}
+
+	if sliceVal.Kind() != reflect.Slice {
```

Comment:
We have a [test](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/cursor_test.go#L95) for the non-pointer case, but we're missing a test for this new error. Can you think of a good way to test this?

Reply:
I've added in a test that hits this case by passing in an interface that is a pointer to a string.